core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1031,6 +1031,8 @@ object SparkSubmit extends CommandLineUtils with Logging {
         } catch {
           case e: SparkUserAppException =>
             exitFn(e.exitCode)
+          case _: Throwable => 
+            exitFn(1)
```

Comment:
This changes will not fix [the problem](https://issues.apache.org/jira/browse/SPARK-34674) completely.
If the main() method completes successfully,  the application will still hang in K8S.
Could you consider also call to exitFn in case of successfull?

Reply:
Yes, you are right.